src/ui/public/visualize/visualize.js

```diff
@@ -8,51 +8,54 @@ import { ResizeCheckerProvider } from 'ui/resize_checker';
 import 'angular-sanitize';
 import './visualization';
 import './visualization_editor';
+import { FilterBarQueryFilterProvider } from 'ui/filter_bar/query_filter';
+
 
 import {
   isTermSizeZeroError,
 } from '../elasticsearch_errors';
 
 uiModules
 .get('kibana/directive', ['ngSanitize'])
-.directive('visualize', function (Notifier, Private, timefilter, getAppState) {
-  const notify = new Notifier({ location: 'Visualize' });
-  const requestHandlers = Private(VisRequestHandlersRegistryProvider);
-  const responseHandlers = Private(VisResponseHandlersRegistryProvider);
-  const ResizeChecker = Private(ResizeCheckerProvider);
-
-  function getHandler(from, name) {
-    if (typeof name === 'function') return name;
-    return from.find(handler => handler.name === name).handler;
-  }
-
-  return {
-    restrict: 'E',
-    scope : {
-      showSpyPanel: '=?',
-      editorMode: '=?',
-      savedObj: '=',
-      appState: '=',
-      uiState: '=?'
-    },
-    template: visualizeTemplate,
-    link: function ($scope, $el) {
-      const resizeChecker = new ResizeChecker($el);
-
-      $scope.vis = $scope.savedObj.vis;
-      $scope.editorMode = $scope.editorMode || false;
-      $scope.vis.editorMode = $scope.editorMode;
-      $scope.vis.visualizeScope = true;
-
-      if (!$scope.appState) $scope.appState = getAppState();
-
-      const requestHandler = getHandler(requestHandlers, $scope.vis.type.requestHandler);
-      const responseHandler = getHandler(responseHandlers, $scope.vis.type.responseHandler);
-
-      $scope.fetch = _.debounce(function () {
+  .directive('visualize', function (Notifier, Private, timefilter, getAppState) {
+    const notify = new Notifier({ location: 'Visualize' });
+    const requestHandlers = Private(VisRequestHandlersRegistryProvider);
+    const responseHandlers = Private(VisResponseHandlersRegistryProvider);
+    const ResizeChecker = Private(ResizeCheckerProvider);
+    const queryFilter = Private(FilterBarQueryFilterProvider);
```

Comment:
i am not really sure about using queryFilter here ...
what about using globalState service to retrieve the filters from there ?